common/network-common/src/main/java/org/apache/spark/network/client/TransportClient.java

```diff
@@ -294,6 +294,15 @@ public void onFailure(Throwable e) {
     }
   }
 
+  /**
+   * Exception thrown when sasl request times out.
+   */
+  public static class SaslTimeoutException extends RuntimeException {
+    public SaslTimeoutException(Throwable cause) {
+      super((cause));
+    }
+  }
+
```

Comment:
Move this to ` org.apache.spark.network.sasl` package.
Also add the common constructors for an Exception - `SaslTimeoutException()`, `SaslTimeoutException(String)`, `SaslTimeoutException(String, Throwable)`, in addition to what we have here.